test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -0,0 +1,258 @@
+def test_init_with_overlap_greater_than_chunk_size():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(split_length=10, split_overlap=15, separators=["."])
+
+
+def test_init_with_invalid_separators():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(separators=[".", 2])
+
+
+def test_init_with_negative_split_length():
+    with pytest.raises(ValueError):
+        _ = RecursiveDocumentSplitter(split_length=-1, separators=["."])
+
+
+def test_apply_overlap_no_overlap():
+    # Test the case where there is no overlap between chunks
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."])
+    chunks = ["chunk1", "chunk2", "chunk3"]
+    result = splitter._apply_overlap(chunks)
+    assert result == ["chunk1", "chunk2", "chunk3"]
+
+
+def test_apply_overlap_with_overlap():
+    # Test the case where there is overlap between chunks
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=4, separators=["."])
+    chunks = ["chunk1", "chunk2", "chunk3"]
+    result = splitter._apply_overlap(chunks)
+    assert result == ["chunk1", "unk1chunk2", "unk2chunk3"]
+
+
+def test_apply_overlap_single_chunk():
+    # Test the case where there is only one chunk
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=3, separators=["."])
+    chunks = ["chunk1"]
+    result = splitter._apply_overlap(chunks)
+    assert result == ["chunk1"]
+
+
+def test_chunk_text_smaller_than_chunk_size():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."])
+    text = "small text"
+    chunks = splitter._chunk_text(text)
+    assert len(chunks) == 1
+    assert chunks[0] == text
+
+
+def test_chunk_text_keep_separator():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."], keep_separator=True)
+    text = "This is a test. Another sentence. And one more."
+    chunks = splitter._chunk_text(text)
+    assert len(chunks) == 3
+    assert chunks[0] == "This is a test."
+    assert chunks[1] == " Another sentence."
+    assert chunks[2] == " And one more."
+
+
+def test_chunk_text_do_not_keep_seperator():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."], keep_separator=False)
+    text = "This is a test. Another sentence. And one more."
+    chunks = splitter._chunk_text(text)
+    assert len(chunks) == 3
+    assert chunks[0] == "This is a test"
+    assert chunks[1] == " Another sentence"
+    assert chunks[2] == " And one more"
+
+
+def test_keep_separator_chunks_are_equal():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."], keep_separator=True)
+    text = "This is a test.This is a test"
+    chunks = splitter._chunk_text(text)
+    assert len(chunks) == 2
+    assert chunks[0] == "This is a test."
+    assert chunks[1] == "This is a test"
+
+
+def test_chunk_text_using_nltk_sentence():
```

Comment:
We should add tests where we use NLTK sentence splitting combined with `split_overlap>0`.